l2l/optimizers/multievolution/optimizer.py

```diff
@@ -117,16 +122,37 @@ def bounding_wrapper(*args, **kwargs):
 
 
     def expand_individual(self, c_population, inner_params):
+        """
+        Expands a multi-individual into many individuals, each with their own parameter combination.
+        This expansion is necessary in order for the evolutionary algorithm to be applied correctly on the parameter
+        space being explored.
+
+        :param  c_population: the population of multi-individuals to be expanded
+        :param inner_params: a description of the parameters which describe each of the individuals within
+        the multi-individual.
+
+        """
         individual_exp = [{} for i in range(len(c_population)*inner_params)]
         for ind_id, elem in enumerate(c_population):
-            for key in self.grouped_params_dict.keys():
-                parameters = elem[key]
-                for ix, e in enumerate(parameters):
-                    individual_exp[ind_id*inner_params+ix][key] = float(e)
-        # print(individual_exp)
+            for ix, (key, e) in enumerate(self.grouped_params_dict.items()):
+                try:
+                    individual_exp[ind_id * inner_params + ix][key] = float(e)
+                except ValueError:
```

Comment:
Here I am not sure if it would work out. If `e` is an array then the casting does not work and a `TypeError`  is thrown. In case `e` is a value, it should work. However, looking at the old code `e` seems to be a value of the list `parameters`, now it is a list. This would also effect storing in `individual_exp`. I suggest to revert these lines to the old code (as that one runs) and try the new code to have a new PR later.